beacon_node/eth2-libp2p/src/peer_manager/mod.rs

```diff
@@ -263,6 +263,53 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         }
     }
 
+    pub fn handle_rpc_error(&mut self, peer_id: &PeerId, protocol: Protocol, err: &RPCError) {
+        // map this error with a `PeerAction` asignated to the peer (if any), and decide if the
+        // peer manager should request some `PeerManagerEvent`
+        let (peer_action, event) = match err {
+            RPCError::IncompleteStream => {
+                // TODO: InvalidMessage? they Clogged? offline?
+                (None, None)
+            }
+            RPCError::InternalError(reason) => {
+                // NOTE: our fault. Now they hate us. Do nothing
+                (None, None)
+            }
+            RPCError::InvalidData => {
+                // TODO: invalid? malicious?
+                (None, None)
+            }
+            RPCError::IoError(e) => {
+                // TODO Whose fault? do nothing?
+                (None, None)
+            }
+            RPCError::RPCErrorResponse => {
+                // TODO what does this mean?
+                (None, None)
+            }
+            RPCError::SSZDecodeError(e) => {
+                // TODO their fault: InvalidMessage
+                (Some(PeerAction::InvalidMessage), None)
+            }
+            RPCError::UnsupportedProtocol => {
+                // TODO: check the protocol and decide
```

Comment:
This one has some nuances. 
The community hasn't decided how to play this one. It would be good if we could get which type of protocol isn't supported. For some protocols (i.e status/ping) we may want to kick them.. or others we may want to be lenient. For the time being I'm leaning towards kicking and banning a peer that doesn't support all protocols, so we force everyone to implement everything. Then later one we can be lenient

Reply:
The protocol is already here, check the signature